src/app/progress-bar/progress-bar.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+
+import { SoldProducts } from '../product';
+import { ProductService } from '../product.service';
+
+@Component({
+  selector: 'app-progress-bar',
+  templateUrl: './progress-bar.component.html',
+  styleUrls: ['./progress-bar.component.scss']
+})
+export class ProgressBarComponent implements OnInit {
+  progressValue: number = 0;
+
+  constructor(private productService: ProductService) { }
+
+  ngOnInit(): void {
+    this.getSoldProducts();
+  }
+
+  processData(soldProducts: SoldProducts):void {
```

Comment:
Never contain any logic in the component. You should think that component is just a way to pass data from service to the view.

Reply:
Logic moved to service